src/main/java/frc/lib/team3061/drivetrain/Drivetrain.java

```diff
@@ -666,10 +666,157 @@ public boolean isMoveToPoseEnabled() {
     return this.isMoveToPoseEnabled;
   }
 
+  // method to convert swerve module number to location
+  private String getSwerveLocation(int swerveModuleNumber) {
+    switch (swerveModuleNumber) {
+      case 0:
+        return "FL";
+      case 1:
+        return "FR";
+      case 2:
+        return "BL";
+      case 3:
+        return "BR";
+      default:
+        return "UNKNOWN";
+    }
+  }
+
   private Command getSystemCheckCommand() {
-    return Commands.sequence(Commands.sequence(Commands.waitSeconds(0.25)))
-        .until(() -> !FaultReporter.getInstance().getFaults(SUBSYSTEM_NAME).isEmpty())
-        .andThen(Commands.runOnce(() -> drive(0, 0, 0, true, false), this));
+    return Commands.sequence(
+        // Tests for driving to the left
+        Commands.parallel(
+            Commands.run(
+                () -> {
+                  io.driveRobotRelative(-1, 0, 0, false);
+                }),
+            Commands.waitSeconds(1)
+                .andThen(
+                    Commands.runOnce(
+                        () -> {
+                          for (int i = 0; i < 4; i++) {
+                            if (inputs.swerve[i].driveVelocityMetersPerSec < 0.5) {
```

Comment:
Great to check that the velocity is greater than some specific value. (We should determine what is a reasonable value, it should be around 3.5 m/s.) However, a given swerve module wheel may be turning in a negative direction if the swerve module is rotated 180° from forward. In addition, we want to make sure that the wheels are all pointed in the right direction. You should write a method that encapsulates all of this:
* make sure that the wheel is within some tolerance of the right direction (or 180° opposite)
* make sure the th wheel is rotating at a velocity greater than some threshold (handling that it may be positive or negative based on the wheel rotation).

Reply:
Encapsulating all of this and adding some parameters to the method can eliminate the repeated code for the next three directions.